httpx/router.go

```diff
@@ -38,7 +41,7 @@ func (r *Router) HandleFunc(path string, f func(context.Context, http.ResponseWr
 
 // Header adds a route that will be used if the header value matches.
 func (r *Router) Headers(pairs ...string) *Route {
-	return &Route{r.mux.Headers(pairs...)}
+	return r.getOrCreateRoute(r.mux.Headers(pairs...), "")
```

Comment:
Hmm, there's an edge case here if you did this I think:

```go
r.Headers("X-Foo", "Bar").Path("/foo").HandlerFunc(Foo)
```

Would it be possible to just re-use the [URLPath](http://www.gorillatoolkit.org/pkg/mux#Route.URLPath) method on `mux.Route`? I _think_ we could do something along the lines of this:

```go
r := RouteFromContext(ctx)
var pairs []string
for k, _ := range Vars(ctx) {
        pairs = append(pairs, k, fmt.Sprinf(":%s", k))
}
return r.URLPath(pairs...)
```